File: src/main/java/com/example/demo/mapper/UaBlacklistRuleMapper.java
package com.example.demo.mapper;

import com.example.demo.model.UaBlacklistRule;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;

import java.util.List;

@Mapper
public interface UaBlacklistRuleMapper {
    List<UaBlacklistRule> selectActive();

    int insert(UaBlacklistRule rule);

    int logicalDelete(@Param("id") Long id);
}

File: src/main/kotlin/com/example/demo/mapper/UaBlacklistRuleMapper.kt
package com.example.demo.mapper

import com.example.demo.model.UaBlacklistRule
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param

@Mapper
interface UaBlacklistRuleMapper {
    fun selectActive(): List<UaBlacklistRule>
    fun insert(rule: UaBlacklistRule): Int
    fun logicalDelete(@Param("id") id: Long): Int
}

